src/pages/map.tsx

```diff
@@ -74,23 +75,21 @@ interface Props {
 
 const MapPage = ({ schedule }: Props) => {
   const [map, setMap] = React.useState<MapboxGl.Map | undefined>(undefined);
-  const [searchBy, setSearchBy] = React.useState<SearchFilter>('kecamatan');
   const [activeLoc, setActiveLoc] = React.useState<LocationData | undefined>(undefined);
-  const [searchKeyword, setSearchKeyword] = React.useState('');
   const { colorMode: mode } = useColorMode();
   const mapFlyoutBackgroundColor = useColorModeValue('white', 'gray.900');
+  const [searchKeyword, setSearchKeyword] = React.useState('');
+  const filtered = useFuzzySearch(schedule, searchKeyword);
 
   const scheduleToRender = () => {
     if (!searchKeyword.length) {
       return schedule;
     }
 
-    const result = schedule.filter(props => {
-      const fieldValue = props[searchBy].toLowerCase();
+    const filteredSchedules = searchKeyword ? filtered.map(s => s.item) : schedule;
 
-      return (
-        fieldValue.includes(searchKeyword.toLowerCase()) && props.detail_lokasi != null && props.detail_lokasi.length
-      );
+    const result = filteredSchedules.filter(props => {
```

Comment:
same with this logic, this can be extracted and reused between index/map pages. TODO for #57